src/Sarif.Converters.UnitTests/TSLintConverterTests.cs

```diff
@@ -0,0 +1,162 @@
+                RuleName = "ruleName.test.value",
+                RuleSeverity = "ruleSeverity.test.value"
+            };
+
+            TSLintLogFix testFix = new TSLintLogFix()
+            {
+                InnerLength = 5,
+                InnerStart = 10,
+                InnerText = "fix.innerText.test.value"
+            };
+
+            testEntry.Fixes = new List<TSLintLogFix>() { testFix };
+
+            TSLintLogPosition testStartPos = new TSLintLogPosition()
+            {
+                Character = 1,
+                Line = 2,
+                Position = 3
+            };
+            testEntry.StartPosition = testStartPos;
+
+            TSLintLogPosition testEndPos = new TSLintLogPosition()
+            {
+                Character = 11,
+                Line = 12,
+                Position = 13
+            };
+            testEntry.EndPosition = testEndPos;
+
+            tsLintLog.Add(testEntry);
+
+            return tsLintLog;
+        }
+
+        // This is necessary because Microsoft.CodeAnalysis.Sarif.Result doesn't implement IEquatable
+        // Must be kept up to date with any changes to TSLintConverter.CreateResult
+        public bool CompareResultToTestData(Result result)
+        {
+            if (!result.RuleId.Equals("ruleName.test.value")) return false;
+            if (!result.Message.Equals("failure.test.value")) return false;
+
+            if (result.Level != ResultLevel.NotApplicable) return false;
+
+            if (result.Locations.Count != 1) return false;
+
+            Region region = result.Locations[0].AnalysisTarget.Region;
+            if (region.StartLine != 3) return false;
+            if (region.StartColumn != 2) return false;
+            if (region.EndLine != 13) return false;
+            if (region.EndColumn != 12) return false;
+
+            if (region.Offset != 3) return false;
+            if (region.Length != 11) return false;
+
+            if (!result.Locations[0].AnalysisTarget.Uri.OriginalString.Equals("name.test.value")) return false;
+
+            if (result.Fixes.Count != 1) return false;
+            if (result.Fixes[0].FileChanges.Count != 1) return false;
+            if (result.Fixes[0].FileChanges[0].Replacements.Count != 1) return false;
+            Replacement replacement = result.Fixes[0].FileChanges[0].Replacements[0];
+
+            if (replacement.Offset != 10) return false;
+            if (replacement.DeletedLength != 5) return false;
+            if (!replacement.InsertedBytes.Equals("fix.innerText.test.value")) return false;
+
+            return true;
+        }
+
+        //[Fact]
+        //[Trait("Category", "E2E")]
+        //public void SimpleE2ETest_ManualValidation()
+        //{
+        //    ToolFormatConverter converter = new ToolFormatConverter();
+        //    converter.ConvertToStandardFormat(ToolFormat.TSLint, @"E:\data\TSLint\TSLint.Results.json", @"E:\data\TSLint\TSLint.Results.sarif");
+        //}
```

Comment:
Commented out (manual verification). I can remove it if you prefer.

Reply:
I see this is fixed.